staging/src/k8s.io/apiserver/pkg/util/flowcontrol/apf_controller.go

```diff
@@ -368,29 +366,28 @@ func newTestableController(config TestableConfig) *configController {
 	return cfgCtlr
 }
 
-func (cfgCtlr *configController) Run(stopCh <-chan struct{}) error {
-	defer utilruntime.HandleCrash()
-
-	// Let the config worker stop when we are done
-	defer cfgCtlr.configQueue.ShutDown()
+func (cfgCtlr *configController) Start(ctx context.Context) error {
+	go func() {
+		<-ctx.Done()
+		// Let the config worker stop when we are done
+		defer cfgCtlr.configQueue.ShutDown()
+	}()
 
 	klog.Info("Starting API Priority and Fairness config controller")
-	if ok := cache.WaitForCacheSync(stopCh, cfgCtlr.plInformerSynced, cfgCtlr.fsInformerSynced); !ok {
-		return fmt.Errorf("Never achieved initial sync")
+	if ok := cache.WaitForCacheSync(ctx.Done(), cfgCtlr.plInformerSynced, cfgCtlr.fsInformerSynced); !ok {
+		return fmt.Errorf("never achieved initial sync")
 	}
 
 	klog.Info("Running API Priority and Fairness config worker")
-	go wait.Until(cfgCtlr.runWorker, time.Second, stopCh)
+	go wait.UntilWithContext(ctx, cfgCtlr.runWorker, time.Second)
 
 	klog.Info("Running API Priority and Fairness periodic rebalancing process")
-	go wait.Until(cfgCtlr.updateBorrowing, borrowingAdjustmentPeriod, stopCh)
+	go wait.UntilWithContext(ctx, cfgCtlr.updateBorrowing, borrowingAdjustmentPeriod)
 
-	<-stopCh
-	klog.Info("Shutting down API Priority and Fairness config worker")
```

Comment:
This log statement could be kept, now in that goroutine forked earlier, right?

Reply:
done

Reply:
Resolved.